iterative/aws/provider.go

```diff
@@ -62,6 +62,32 @@ func ResourceMachineCreate(ctx context.Context, d *schema.ResourceData, m interf
 	}
 	svc := ec2.NewFromConfig(config)
 
+	// Availability zone
+	if availabilityZone == "" {
```

Comment:
Small comment: this could be moved to after line 247, as it's only needed if subnetId is not provided.

Reply:
https://github.com/iterative/terraform-provider-iterative/pull/670/commits/729aa586bb8fef149733a1a93ecb667e66d2f9ae